add exp to transforms

mdmould opened the pull request:
Implementation of exponential transform. Useful for, e.g., transforming from $\mathbb{R}$ to a one-side bounded domain $(a, \infty)$ or $(-\infty, a)$ via composition with an affine transform.

arturbekasov commented:
Hey @mdmould,

Thanks for taking the time to open the PR.

The implementation LGTM. Could you also add `Exp` to [NonlinearitiesTest](https://github.com/bayesiains/nflows/blob/master/tests/transforms/nonlinearities_test.py#L99)?

Cheers,

Artur

mdmould commented:
Hi @arturbekasov,

I've added Exp in the three lists of transforms in NonlinearitiesTest. It looks like the transforms are only called on numbers in $(0,1)$ so the test_inverse should be fine. I also added a domain test for the inverse, similar to TanhTest.

Cheers,
Matt
